lib/src/gotrue_admin_api.dart

```diff
@@ -64,8 +64,16 @@ class GoTrueAdminApi {
   /// Get a list of users.
   ///
   /// This function should only be called on a server. Never expose your `service_role` key on the client.
-  Future<List<User>> listUsers() async {
-    final options = GotrueRequestOptions(headers: _headers);
+  ///
+  /// The result is paginated. Use the [page] and [perPage] parameters to paginate the result.
+  Future<List<User>> listUsers({int? page, int? perPage}) async {
+    final options = GotrueRequestOptions(
+      headers: _headers,
+      body: {
+        if (page != null) 'page': page,
+        if (perPage != null) 'per_page': perPage,
+      },
```

Comment:
It seems like the parameters are being sent as query parameters here!
https://github.com/supabase/gotrue-js/blob/master/src/GoTrueAdminApi.ts#L167

```suggestion
      query: {
        if (page != null) 'page': page.toString(),
        if (perPage != null) 'per_page': perPage.toString(),
      },
```

Reply:
@Vinzent03 
Great point!

I just checked, and it seems like null value and empty string are being treated the same way in the backend. [Here](https://github.com/supabase/gotrue/blob/master/api/pagination.go#L42) is where the query parameters are being parsed, and `params.Get("per_page")` returns an empty string if the query parameter `per_page` is not present.